lib/galaxy/tools/parser/interface.py

```diff
@@ -148,6 +148,17 @@ def parse_requirements_and_containers(self):
     def parse_input_pages(self):
         """ Return a PagesSource representing inputs by page for tool. """
 
+    def parse_provided_metadata_style(self):
+        """Return style of tool provided metadata file (e.g. galaxy.json).
+
+        Return "legacy" for older pre-17.09 default.
+        """
+        return "default"
```

Comment:
So classes that subclass get this as a default ? I.e is it intentional for YamlToolSource that is in the XmlToolSource ?